### PR TITLE
add cursor_ray example

### PR DESCRIPTION
**Pull request by alisterburt**
https://user-images.githubusercontent.com/7307488/125622656-41874f54-746d-42e6-89dd-60e640b75475.mp4

### CODE REVIEW

File: examples/cursor_ray.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+ray_layer = viewer.add_points(
+    points, face_color=point_colors, size=point_sizes, name='cursor ray'
+)
+volume_layer = viewer.add_image(volume, blending='additive')
+
+
+# callback function, called on mouse click when volume layer is active
+@volume_layer.mouse_drag_callbacks.append
+def on_click(layer, event):
+    world_to_canvas = viewer.window.qt_viewer.view.scene.transform
```
Comment (alisterburt) [inline examples/cursor_ray.py:50]:
```suggestion
def on_click(layer):
```

File: examples/cursor_ray.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+)
+ray_layer = viewer.add_points(
+    points, face_color=point_colors, size=point_sizes, name='cursor ray'
+)
+volume_layer = viewer.add_image(volume, blending='additive')
+
+
+# callback function, called on mouse click when volume layer is active
+@volume_layer.mouse_drag_callbacks.append
+def on_click(layer):
```
Comment (alisterburt) [inline examples/cursor_ray.py:49]:
```suggestion
def on_click(layer, event):
```

File: examples/cursor_ray.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    near_point, far_point = layer._cursor_ray(event, world_to_canvas)
```
Comment (alisterburt) [inline examples/cursor_ray.py:50]:
```suggestion
    near_point, far_point = layer._cursor_ray(event)
```

**Review by kevinyamauchi — APPROVED** [review]
Amazing!

**Pull request merged**